src/modules/Icon/Icon.tsx

```diff
@@ -32,37 +33,27 @@ const Icon: React.FC<IProps> = ({
   size
 }) => {
   const theme = useTheme();
-  const svgIcon = src || (theme && `assets/icons/${theme.icons[name]}`);
   const [styles, cx] = useStyles(styleSheet);
-  return (
-    <ReactSVG
-      src={svgIcon}
-      className={cx(styles.icon)}
-      beforeInjection={svg => {
-        svg.setAttribute("class", "y-icon");
-        svg.setAttribute("class", `${svg.getAttribute("class")} ${className}`);
-        if (size) {
-          svg.setAttribute("height", size.toString());
-          svg.setAttribute("width", size.toString());
-        }
-        if (height) {
-          svg.setAttribute("height", height.toString());
-        }
-        if (width) {
-          svg.setAttribute("width", width.toString());
-        }
-        if (stroke) {
-          svg.setAttribute("stroke", stroke);
-        }
-        if (fill) {
-          svg.setAttribute("fill", fill);
-        }
-        if (color) {
-          svg.setAttribute("fill", color);
-          svg.setAttribute("stroke", color);
-        }
-      }}
-    />
-  );
+
+  const SvgIcon: any = useMemo(() => {
+    return src || theme.icons[name] || "";
+  }, [src, theme, name]);
+
+  const newProps: any = {};
+  if (size || width) {
+    newProps.width = size || width;
+  }
+  if (size || height) {
+    newProps.height = size || height;
+  }
+  if (fill || color) {
+    newProps.fill = fill || color;
+  }
+  if (stroke || color) {
+    newProps.stroke = stroke || color;
+  }
+  newProps.className = `y-icon ${cx(styles.icon)} ${className || ""}`;
+
+  return <SvgIcon {...newProps} />;
 };
-export default Icon;
+export default withStyles(() => ({}))(Icon);
```

Comment:
Maybe we could pass the stylesheet here instead of using the hook? Even if just for consistency sake, since all (most?) other components do it